apps/webapp/app/v3/environmentVariables/environmentVariablesRepository.server.ts

```diff
@@ -0,0 +1,505 @@
+import { Prisma, PrismaClient } from "@trigger.dev/database";
+import { $transaction, prisma } from "~/db.server";
+import { getSecretStore } from "~/services/secrets/secretStore.server";
+import { EnvironmentVariable, ProjectEnvironmentVariable, Repository, Result } from "./repository";
+import { z } from "zod";
+
+function secretKeyProjectPrefix(projectId: string) {
+  return `environmentvariable:${projectId}:`;
+}
+
+function secretKeyEnvironmentPrefix(projectId: string, environmentId: string) {
+  return `${secretKeyProjectPrefix(projectId)}${environmentId}:`;
+}
+
+function secretKey(projectId: string, environmentId: string, key: string) {
+  return `${secretKeyEnvironmentPrefix(projectId, environmentId)}${key}`;
+}
+
+function parseSecretKey(key: string) {
+  const parts = key.split(":");
+  return {
+    projectId: parts[1],
+    environmentId: parts[2],
+    key: parts[3],
+  };
+}
+
+export class EnvironmentVariablesRepository implements Repository {
+  constructor(private prismaClient: PrismaClient = prisma) {}
+
+  async create(
+    projectId: string,
+    userId: string,
+    options: { key: string; values: { value: string; environmentId: string }[] }
+  ): Promise<Result> {
+    const project = await this.prismaClient.project.findUnique({
+      where: {
+        id: projectId,
+        organization: {
+          members: {
+            some: {
+              userId,
+            },
+          },
+        },
+        deletedAt: null,
+      },
+      select: {
+        environments: {
+          select: {
+            id: true,
+          },
+        },
+      },
+    });
+
+    if (!project) {
+      return { success: false as const, error: "Project not found" };
+    }
+
+    if (options.values.every((v) => !project.environments.some((e) => e.id === v.environmentId))) {
+      return { success: false as const, error: `Environment not found` };
+    }
+
+    //get rid of empty strings
+    const values = options.values.filter((v) => v.value.trim() !== "");
+
+    if (values.length === 0) {
+      return { success: false as const, error: `You must set at least one value` };
+    }
+
+    try {
+      const result = await $transaction(this.prismaClient, async (tx) => {
+        const environmentVariable = await tx.environmentVariable.create({
+          data: {
+            key: options.key,
```

Comment:
environment variables should have a friendlyId, in case we do stuff with them in our API eventually